server/data/data-library/src/main/java/org/apache/james/rrt/lib/MappingsImpl.java

```diff
@@ -104,38 +104,17 @@ public Builder add(Mapping mapping) {
             mappings.add(mapping);
             return this;
         }
-
         
         public Builder addAll(Mappings mappings) {
             this.mappings.addAll(mappings);
             return this;
         }
         
         public MappingsImpl build() {
-            Comparator<? super Mapping> byTypeComparator = (mapping1, mapping2) -> {
-                if (mapping1.getType() == mapping2.getType()) {
-                    return mapping1.asString().compareTo(mapping2.asString());
-                }
-                if (mapping1.getType() == Type.Domain) {
-                    return -1;
-                }
-                if (mapping2.getType() == Type.Domain) {
-                    return 1;
-                }
-                if (mapping1.getType() == Type.Forward) {
-                    return -1;
-                }
-                if (mapping2.getType() == Type.Forward) {
-                    return 1;
-                }
-                return mapping1.asString().compareTo(mapping2.asString());
-            };
-            return new MappingsImpl(mappings.build().stream()
-                    .sorted(byTypeComparator)
-                    .collect(Guavate.toImmutableList()));
+            return new MappingsImpl(mappings.build())
+                .sort();
```

Comment:
this is a very bad idea, it's a temporal coupling : you don't know, when you receive a Mappings if it's sorted or not without knowing the origin of your instance.
Having it sorted as an invariant is much better.